file-formats/ints/type/zif_aff_ints_v1.intf.abap

```diff
@@ -0,0 +1,278 @@
+      "! <p class="shorttext">Success</p>
+      "! Success message
+      success     TYPE ty_object_status VALUE 'S',
+    END OF co_object_status.
+
+  "! $values { @link zif_aff_ints_v1.data:co_logical_object_type }
+  "! $default { @link zif_aff_ints_v1.data:co_logical_object_type.cds_view }
+  TYPES ty_logical_object_type TYPE c LENGTH 1.
+  CONSTANTS:
+    BEGIN OF co_logical_object_type,
+      "! <p class="shorttext">Table Function</p>
+      "! Table Function
+      table_function TYPE ty_logical_object_type VALUE 'F',
+      "! <p class="shorttext">CDS View</p>
+      "! CDS view
+      cds_view       TYPE ty_logical_object_type VALUE 'V',
+      "! <p class="shorttext">Class</p>
+      "! Class
+      class          TYPE ty_logical_object_type VALUE 'C',
+      "! <p class="shorttext">DCL</p>
+      "! DCL
+      dcl            TYPE ty_logical_object_type VALUE 'D',
+      "! <p class="shorttext">Model</p>
+      "! Model
+      model          TYPE ty_logical_object_type VALUE 'M',
+      "! <p class="shorttext">Avas</p>
+      "! Avas
+      avas           TYPE ty_logical_object_type VALUE 'A',
+    END OF co_logical_object_type.
+
+  TYPES:
+    "! <p class="shorttext">Scenario Object</p>
+    "! Object generated based on a scenario
+    BEGIN OF ty_scenario_ddl_object,
+      "! <p class="shorttext">Object Name</p>
+      "! Object name
+      object_name         TYPE c LENGTH 40,
+      "! <p class="shorttext">Object Type</p>
+      "! Type of the object
+      object_type         TYPE c LENGTH 4,
+      "! <p class="shorttext">Object Status</p>
+      "! Status of the object
+      object_status       TYPE ty_object_status,
+      "! <p class="shorttext">Logical Object Type</p>
+      "! Logical object type of a BOM object
+      logical_object_type TYPE ty_logical_object_type,
+    END OF ty_scenario_ddl_object,
+
+    "! <p class="shorttext">Intelligent Scenario</p>
+    "! Intelligent scenario
+    BEGIN OF ty_intelligent_scenario,
+      "! <p class="shorttext">Intelligent Scenario Type</p>
+      "! Intelligent scenario type
+      "! $required
+      scenario_type              TYPE c LENGTH 256,
+      "! <p class="shorttext">Extensible</p>
+      "! If true, the scenario is extensible
+      is_extensible              TYPE abap_bool,
+      "! <p class="shorttext">Multiple Active Models</p>
+      "! If true, the scenario can have multiple active models
+      is_multi_active_model      TYPE abap_bool,
+      "! <p class="shorttext">Data Management</p>
+      "! If true, the scenario uses Data Management
+      is_data_management_enabled TYPE abap_bool,
+      "! <p class="shorttext">Automate Turnkey Switch On</p>
+      "! If true, scenario will be turnkey enabled to automate the operations
+      is_turnkey_auto_switch_on  TYPE abap_bool,
+      "! <p class="shorttext">Turnkey Implementation Type</p>
+      "! Type of turnkey implementation
+      turnkey_type               TYPE c LENGTH 10,
```

Comment:
I saw you made a value help for this field. Can we also have an enum here like we did for object_statusand logical_object_type?

Reply:
In future, we may call backend class method to get the list of possible values. So, enum is not possible in that case. Also, this is still under discussion.